src/CMakeLists.txt

```diff
@@ -34,8 +34,12 @@ endif ()
 
 # Build the C library
 add_subdirectory (clib)
+set(CFLAGS ${CFLAGS}  PARENT_SCOPE)
+set(CPPFLAGS ${CPPFLAGS} PARENT_SCOPE)
```

Comment:
Can you tell me why you moved these here? I'm trying to better understand cmake...

Reply:
Variables defined in src/clib aren't available to the cmake in parent directories without setting with PARENT_SCOPE
and even then only one level up.   Rather than sending up CMAKE_C_FLAGS and CMAKE_C_FLAGS_${CMAKE_BUILD_TYPE} I combined these into CFLAGS in src/clib/CMakeLists.xml and sent CFLAGS up to the top level so that the libpio.settings file could be filed in.